### PR TITLE
GRCC: move into grc tree and rewrite

### PR DESCRIPTION
**Pull request by skoslowski**
This a complete overhaul of grcc. Also moves from gr-utils to grc. This enables grcc to

- use the auto generation of depending grc hier blocks
- respect custom run commands

There is also some simple pytest code in there

### CODE REVIEW

**Comment by jmcorgan** [comment]
You've assigned it to Seth and marked it under review; does that indicate he'll do a review on this and indicate approval before merge, or is this ready to go?

**Comment by skoslowski** [comment]
Yes, Seth will do a review first.

**Comment by jmcorgan** [comment]
@sdh11 please update with status

**Comment by sdh11** [comment]
Reviewing today.

File: grc/compiler.py [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+from .core.Platform import Platform
+
+
+def argument_parser():
+    parser = argparse.ArgumentParser(description=(
+        "Compiles a GRC file (.grc) into a GNU Radio Python program. "
+        "The program is stored in ~/.grc_gnuradio by default, "
+        "but this location can be changed with the -d option."
+    ))
+    parser.add_argument("-d", "--directory", dest='out_dir', default='',
```
Comment (sdh11) [inline grc/compiler.py]: The output should default to the current directory.

**Comment by skoslowski** [comment]
This commit changes the CLI as discussed (before I tried to keep compatibility with older versions)
We now have:
```
usage: grcc [-h] [-o DIR] [-u] [-r] GRC_FILE [GRC_FILE ...]

Compile a GRC file (.grc) into a GNU Radio Python program and run it.

positional arguments:
  GRC_FILE              .grc file to compile

optional arguments:
  -h, --help            show this help message and exit
  -o DIR, --output DIR  Output directory for compiled program [default=.]
  -u, --user-lib-dir    Output to default hier_block library (overwrites -o)
  -r, --run             Run the program after compiling [default=False]
```

**Pull request merged**